### DISCUSSION TITLE
Warnings, messages & mismatch

### DISCUSSION THREAD

**Discussion by kchoX14** [threeML/threeML#478] [Q&A]
Hi!

I installed threeML using ```pip``` on my Ubuntu 18.04 LTS platform. I printed the configuration in line 2 with 

```
from threeML import threeML_config, show_configuration

show_configuration()
```

It showed me a few warnings before printing out the result. 

```
[WARNING ] The naima package is not available. Models that depend on it will not be available
[WARNING ] The GSL library or the pygsl wrapper cannot be loaded. Models that depend on it will not be available.
[WARNING ] The ebltable package is not available. Models that depend on it will not be available
[INFO    ] Starting 3ML!
[WARNING ] ROOT minimizer not available
[WARNING ] Multinest minimizer not available
[WARNING ] PyGMO is not available
[WARNING ] The cthreeML package is not installed. You will not be able to use plugins which require the C/C++ interface (currently HAWC)
[WARNING ] Could not import plugin FermiLATLike.py. Do you have the relative instrument software installed and configured?
[WARNING ] Could not import plugin HAWCLike.py. Do you have the relative instrument software installed and configured?
[WARNING ] Env. variable OMP_NUM_THREADS is not set. Please set it to 1 for optimal performances in 3ML
[WARNING ] Env. variable MKL_NUM_THREADS is not set. Please set it to 1 for optimal performances in 3ML
[WARNING ] Env. variable NUMEXPR_NUM_THREADS is not set. Please set it to 1 for optimal performances in 3ML
CONFIG
.
(contd.)
.
```

Is it normal and safe to ignore?

--------------------------------------------------------------------------------------------------------------

EDIT no. 4: 
1. There is an ```emcee``` version incompatibility between ```naima``` and ```threeml```. So, one isn't able to function with the ```emcee``` of the other due to version inconsistencies (checked with ```pip install```). ```pip``` v21.2.2, ```python``` v3.8.10.
- For **threeML v2.2.2**:   ```emcee version >= 3```
- For **naima v0.9.1**:   ```2.1.1 <= emcee version < 3``` 
2. I could only ```pip install``` PyGMO and ebltable successfully. I also set the three environment variables in my root ```.bashrc``` script with as **export <varname>=1**. I restarted my Python script over ```spyder``` (v5.0.5) and ran the commands again. While the ```naima```, ```pygmo``` and ```ebltable``` warnings vanished, the env. variables ones still persist.

**Comment by grburgess** [comment]
Warnings and startup warnings are meant to inform you of things that could be lead to issues. The description of the warning will let you know if you need to be concerned. If there is an actual issue, an error will be thrown, halting the execution of the program.

There is a version mismatch with name's requirements for emcee (#435 ). After installing naima (if it is needed) one can reinstall emcee for the up to date version required by 3ML. 

Not all packages are required for 3ML to work. Only those required for an analysis.
